kimchi/src/precomputed_srs.rs

```diff
@@ -0,0 +1,82 @@
+//! To prover and verify proofs you need a [Structured Reference String](https://www.cryptologie.net/article/560/zk-faq-whats-a-trusted-setup-whats-a-structured-reference-string-whats-toxic-waste/) (SRS).
+//! The generation of this SRS is quite expensive, so we provide a pre-generated SRS in this repo.
+//! Specifically, two of them, one for each pasta curve.
+//!
+//! We generate the SRS within the test in this module.
+//! If you modify the SRS, you will need to regenerate the SRS by passing the `SRS_OVERWRITE` env var.
+
+use std::path::PathBuf;
+use std::{fs::File, io::BufReader};
+
+use poly_commitment::srs::SRS;
+
+use crate::curve::KimchiCurve;
+
+/// The size of the SRS that we serialize.
+pub const SERIALIZED_SRS_SIZE: u32 = 16;
+
+/// The path of the serialized SRS.
+fn get_srs_path<G: KimchiCurve>() -> PathBuf {
+    let base_path = std::env::var("CARGO_MANIFEST_DIR").expect("failed to get manifest path");
+    PathBuf::from(base_path)
+        .join("../srs")
+        .join(format!("{}.srs", G::NAME))
+}
+
+/// Obtains an SRS for a specific curve from disk.
+/// Panics if the SRS does not exists.
```

Comment:
Given than it isn't an unrecoverable state, wouldn't it be better to return `Option` or `Result`?
Now if you don't want to panic the only way is to check the file beforehand, but I think it would be better if it didn't panic and if you want it to panic you can still use `unwrap`.

Reply:
I couldn't think of a case where you would not want to panic here, as it's probably a configuration issue and you'll realize that from the start. The situation is different now if you can request an SRS of a specific size or something :o

Reply:
One case I can think of is "create if it doesn't exist".
But the main issue is that you go from result to panic, but not the other way around. If an error isn't unrecoverable, a library should ideally return a result and let the users decide if they should panic.
I also just noticed that it calls `get_srs_path` that can panic, those panics should also be the doc if we keep the panic approach.